src/include/daos_srv/dtx_srv.h

```diff
@@ -214,13 +214,15 @@ int dtx_obj_sync(uuid_t po_uuid, uuid_t co_uuid, struct ds_cont_child *cont,
 int dtx_handle_resend(daos_handle_t coh, struct dtx_id *dti,
 		      daos_epoch_t *epoch, uint32_t *pm_ver);
 
-/* XXX: The higher 48 bits of HLC is the wall clock, the lower bits are for
- *	logic clock that will be hidden when divided by NSEC_PER_SEC.
- */
 static inline uint64_t
 dtx_hlc_age2sec(uint64_t hlc)
 {
-	return (crt_hlc_get() - hlc) / NSEC_PER_SEC;
+	uint64_t now = crt_hlc_get();
+
+	if (now <= hlc)
```

Comment:
The issue is that if the client side HLC does not sync with server side HLC any longer, then we should not use the HLC inside the DTX ID to check the RPC age for dtx_handle_resend().

Reply:
Sorry, only see this now. HLC sync only guarantees that crt_hlc_get() > hlc. The age calculation is only meaningful if physical clocks are reasonably synchronized (which != HLC sync).